### PR TITLE
✨ Add WebEngage as an analytics vendor

### LINKED ISSUE #22124: Intent to Implement: WebEngage
We at [WebEngage](https://webengage.com/) would love to add our platform as an amp-analytics plugin, we already created a new API which supports query string and have everything ready with us.

A simple example that sends us a pageview:

```
<amp-analytics id="webengage-amp-analytics" >
    <script type="application/json">
    {
      "vars": {
        "licenseCode": "~134105385"
      },
      "requests": {
        "base": "https://c.webengage.com/amp?licenseCode=${licenseCode}&luid=${clientId(we_luid)}&pageUrl=${canonicalUrl}&pageTitle=${title}&referrer=${documentReferrer}&vh=${viewportHeight}&vw=${viewportWidth}",
        "we_amp_pageview_request": {
            "baseUrl": "${base}&eventName=Page Viewed&category=application"
        }
      },
      "triggers": {
        "we_amp_pageview": {
            "on": "visible",
            "request": "we_amp_pageview_request"
        }
      }
    }
    </script>
  </amp-analytics>
```

we would ask our customers to add more triggers/request if they need to send us any other amp-analytics supported events.

Do let us know feedback/issue if any, before I open a PR.

### PR DESCRIPTION
**Pull request by ayushgupta29**
Resolves #22124 . Adds WebEngage as an amp-analytics vendor option.
@lannka

### CODE REVIEW

**Comment by torch2424** [comment]
cc @calebcordry and @zhouyx in case I don't get to this while in Berlin :)

File: examples/analytics-vendors.amp.html [hunk 1]
```diff
@@ -1598,6 +1599,29 @@
...
+        "weClickTrigger": {
+          "on": "click",
+          "selector": "#test1",
+          "request": "weClick"
+        }
+      }
+    }
+  </script>
+</amp-analytics>
+<!-- -->
```
Comment (torch2424) [inline examples/analytics-vendors.amp.html]: nit: Can you fill this out, or remove it? 😄

File: examples/analytics-vendors.amp.html [hunk 2]
```diff
@@ -1598,6 +1599,29 @@
   </script>
 </amp-analytics>
 <!-- Vpon Analytics example -->
+<!-- start WebEngage Analytics example -->
```
Comment (torch2424) [inline examples/analytics-vendors.amp.html]: nit: Can you add a newline before here? 😄

File: extensions/amp-analytics/0.1/vendors/webengage.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
```
Comment (torch2424) [inline extensions/amp-analytics/0.1/vendors/webengage.js]: Wrong Copyright year 😄

Comment (ayushgupta29) [reply extensions/amp-analytics/0.1/vendors/webengage.js]: @torch2424 done.

**Comment by torch2424** [comment]
@ayushgupta29 Thank you for making the changes! After looking at the tests: https://travis-ci.org/ampproject/amphtml/jobs/540193477

Seems like something is wrong with your json config, could you please look into that? Thank you! 😄

**Comment by ayushgupta29** [comment]
@torch2424 I couldn't see any reason for the tests getting failed. I tested the request using the example page, and everything seem to be in order. Could you please help me in reviewing the errors?

**Comment by torch2424** [comment]
@ayushgupta29

So on the tests page: https://travis-ci.org/ampproject/amphtml/jobs/541201715

It has the following:

Screen Shot 2019-06-04 at 3 55 29 PM
Screen Shot 2019-06-04 at 3 55 42 PM


So I would suggest running: 'gulp test --local-changes --watch' to help debug this 😄 

Also, cc @zhouyx to see if they also have some suggestions.

Thanks! 😄 👍

**Comment by torch2424** [comment]
@ayushgupta29

Let me know when this is good to go, and do another round of review. Thanks! 😄

**Comment by zhouyx** [comment]
I don't have a good idea why the tests are failing with a quick look. @torch2424 if it fails continuously, we can run locally to take a look. Thanks.

**Comment by ayushgupta29** [comment]
@zhouyx @torch2424 I am unable to figure out why the tests are failing. Would you be able to take a look in local?

File: extensions/amp-analytics/0.1/test/vendor-requests.json [hunk 4]
```diff
@@ -450,6 +450,10 @@
     "elementInteract": "https://tags-dmp.vpadn.com/et?t=_timestamp_&sdkn=j&sdkv=1.2.0&lk=!licence_key&en=UTF-8&ctid=_client_id_&ev=element_interact&pl={\"name\":\"!category\",\"action\":\"!action\",\"value\":\"!label\"}&is_amp=1&page_id=_page_view_id_",
     "event": "https://tags-dmp.vpadn.com/et?t=_timestamp_&sdkn=j&sdkv=1.2.0&lk=!licence_key&en=UTF-8&ctid=_client_id_&ev=!ev_name&pl=!payload&is_amp=1&page_id=_page_view_id_"
   },
+  "webengage": {
+    "base": "https://c.webengage.com/amp?licenseCode=!licenseCode&luid=_client_id_&pageUrl=_canonical_url_&pageTitle=_title_&referrer=_document_referrer_&vh=_viewport_height_&vw=_viewport_width_&category=application",
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/test/vendor-requests.json]: The requests will be resolved with the `extraUrlParams` you specified. `....&isAmp=1` that will fix the test.

**Comment by ayushgupta29** [comment]
@torch2424 @torch2424 Thanks for the help. The tests have passed. Let me know if anything else is required from my end.

I will add a documentation as well in the respective repo.

**Comment by torch2424** [comment]
@ayushgupta29

Thanks again! And as you stated:

> I will add a documentation as well in the respective repo.

Thank you for mentioning this 😄 Let us know if you need help with that!

**Comment by ayushgupta29** [comment]
@torch2424 Has this been released? I don't see webengage config being loaded in https://cdn.ampproject.org/v0/amp-analytics-0.1.js

**Comment by torch2424** [comment]
So our release cycle is once every two weeks (1 week in canary, and then the second goes to PROD).

So since this was merged about a week ago, It'll probably go out sometime next week wednesday 😄 

Thank you!

**Pull request merged**